tests/unit_test/events/definitions_test.py

```
@@ -1199,3 +1198,136 @@
+        except AppException as e:
+            pytest.fail(f"Unexpected exception: {e}")
+
+    @responses.activate
+    def test_trigger_mail_channel_process_event_execute_exception(self):
+        from kairon.events.definitions.mail_channel import MailProcessEvent
+        from kairon.exceptions import AppException
+        from unittest.mock import patch
+
+        with patch("kairon.shared.channels.mail.processor.MailProcessor.process_message_task",
+                   side_effect=Exception("Test")):
+            with pytest.raises(AppException, match="Test"):
```

Ruff (0.8.0)

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)